src/backend/base/langflow/api/v1/mcp.py

```diff
@@ -0,0 +1,295 @@
+    """Generate JSON schema from flow input nodes."""
+    # Get the flow's data which contains the nodes and their configurations
+    flow_data = flow.data if flow.data else {}
+
+    graph = Graph.from_payload(flow_data)
+    input_nodes = [vertex for vertex in graph.vertices if vertex.is_input]
+
+    properties = {}
+    required = []
+    for node in input_nodes:
+        node_data = node.data["node"]
+        template = node_data["template"]
+
+        for field_name, field_data in template.items():
+            if field_data != "Component" and field_data.get("show", False) and not field_data.get("advanced", False):
+                field_type = field_data.get("type", "string")
+                properties[field_name] = {
+                    "type": field_type,
+                    "description": field_data.get("info", f"Input for {field_name}"),
+                }
+                # Update field_type in properties after determining the JSON Schema type
+                if field_type == "str":
+                    field_type = "string"
+                elif field_type == "int":
+                    field_type = "integer"
+                elif field_type == "float":
+                    field_type = "number"
+                elif field_type == "bool":
+                    field_type = "boolean"
+                else:
+                    logger.warning(f"Unknown field type: {field_type} defaulting to string")
+                    field_type = "string"
+                properties[field_name]["type"] = field_type
+
+                if field_data.get("required", False):
+                    required.append(field_name)
+
+    return {"type": "object", "properties": properties, "required": required}
+
+
+@server.list_prompts()
+async def handle_list_prompts():
+    return []
+
+
+@server.list_resources()
+async def handle_list_resources():
+    return []
+
+
+@server.list_tools()
+async def handle_list_tools():
+    try:
+        session = await anext(get_session())
+        flows = (await session.exec(select(Flow))).all()
+        tools = []
+        name_count = {}  # Track name occurrences
+
+        for flow in flows:
+            if flow.user_id is None:
+                continue
+            # Generate unique name by appending _N if needed
+            base_name = flow.name
+            if base_name in name_count:
+                name_count[base_name] += 1
+                unique_name = f"{base_name}_{name_count[base_name]}"
+            else:
+                name_count[base_name] = 0
+                unique_name = base_name
+
+            tool = types.Tool(
+                name=str(flow.id),  # Use flow.id instead of name
+                description=f"{unique_name}: {flow.description}"
+                if flow.description
+                else f"Tool generated from flow: {unique_name}",
+                inputSchema=json_schema_from_flow(flow),
+            )
+            tools.append(tool)
+
+        return tools
```

Comment:
Names are unique per user. What if we added the id like `name ({id})`?

Reply:
actually ended up using the flow.id as the unique name so let's take out the _N stuff

Reply:
But it would be good for the user to see the name, wouldn't it?